sky/unit/test/widget/gesture_detector_test.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:sky/widgets.dart';
+import 'package:test/test.dart';
+
+import '../engine/mock_events.dart';
+import 'widget_tester.dart';
+
+void main() {
+  test('Uncontested scrolls start immediately', () {
+    WidgetTester tester = new WidgetTester();
+
+    bool didStartScroll = false;
+    double updatedScrollDelta;
+    bool didEndScroll = false;
+
+    Widget builder() {
+      return new GestureDetector(
+        onVerticalScrollStart: () {
+          didStartScroll = true;
+        },
+        onVerticalScrollUpdate: (double scrollDelta) {
+          updatedScrollDelta = scrollDelta;
+        },
+        onVerticalScrollEnd: () {
+          didEndScroll = true;
+        },
+        child: new Container()
+      );
+    }
+
+    tester.pumpFrame(builder);
+    expect(didStartScroll, isFalse);
+    expect(updatedScrollDelta, isNull);
+    expect(didEndScroll, isFalse);
+
+    Point firstLocation = new Point(10.0, 10.0);
+    tester.dispatchEvent(tester.pointerGenerator.down(7, firstLocation), firstLocation);
+    expect(didStartScroll, isTrue);
+    didStartScroll = false;
+    expect(updatedScrollDelta, isNull);
+    expect(didEndScroll, isFalse);
+
+    Point secondLocation = new Point(10.0, 9.0);
+    tester.dispatchEvent(tester.pointerGenerator.move(7, secondLocation), secondLocation);
+    expect(didStartScroll, isFalse);
+    expect(updatedScrollDelta, 1.0);
+    updatedScrollDelta = null;
+    expect(didEndScroll, isFalse);
+
+    tester.dispatchEvent(tester.pointerGenerator.up(7), secondLocation);
+    expect(didStartScroll, isFalse);
+    expect(updatedScrollDelta, isNull);
+    expect(didEndScroll, isTrue);
+    didEndScroll = false;
+
+    tester.pumpFrame(() => new Container());
+  });
+}
```

Comment:
do you have a test for the reverse? that a contested scroll doesn't start immediately?